core/src/main/java/com/yahoo/imapnio/async/response/ImapResponseMapper.java

```diff
@@ -96,6 +121,30 @@ public ImapResponseMapper() {
         if (valueType == SearchResult.class) {
             return (T) parser.parseToSearchResult(content);
         }
+
+        return readValue(content, valueType, EMPTY_FETCH_ITEMS);
+    }
+
+    /**
+     * Method to deserialize IMAPResponse content from given IMAPResponse content String.
+     *
+     * @param <T> the object to serialize to
+     * @param content list of IMAPResponse obtained from server.
+     * @param valueType class name that this api will convert to.
+     * @param extensionItems the array of extension FetchItem
+     * @return the serialized object
+     * @throws ImapAsyncClientException when target class to covert to is not supported
+     */
+    @Nonnull
+    public <T> T readValue(@Nonnull final IMAPResponse[] content, @Nonnull final Class<T> valueType, @Nonnull final FetchItem[] extensionItems)
```

Comment:
I cannot merge this because Fetch Response may accept extension and there is no way to know the details of fetch response extension item class structure without passing in an extra field.

Reply:
I dont think store responses returns extension.  Could u please check?